spec/features/create_preassembly_image_spec.rb

```diff
@@ -68,13 +68,10 @@
     # go to job details page, download result
     first('td > a').click
     expect(page).to have_content preassembly_project_name
+
     # wait for preassembly background job to finish
-    Timeout.timeout(Settings.timeouts.workflow) do
-      loop do
-        page.evaluate_script('window.location.reload()')
-        break if page.has_link?('Download')
-      end
-    end
+    reload_page_until_timeout!(text: 'Download')
```

Comment:
note that `.has_link?` is not the same as `text:`

Reply:
@ndushay I believe link text matches on `has_text` too.  I'm not sure that it being a link is critical to the purpose of this test.